tests/rest/auth/test_oidc.py

```diff
@@ -437,46 +437,106 @@ def test_oidc_device_flow_without_pkce_nor_secret(requests_mock, caplog):
         flags=re.DOTALL
     )
 
-
-@pytest.mark.parametrize(["mode", "client_id", "use_pkce", "client_secret", "expected_fields"], [
-    (
+# fmt: off
+@pytest.mark.parametrize(
```

Comment:
FYI: instead of one `@pytest.mark.parametrize` huge listing, it's also possible to combine multiple (which are combined in the same way as `itertools.product`). This will make this parameterization more compact I think

```
@pytest.mark.parametrize(["mode", "client_id", "use_pkce", "client_secret", "expected_fields"], [ ....
@pytest.mark.parametrize("use_verification_uri_complete", [False, True]
def test....
```

Reply:
That was exactly the kind of thing I was looking for. Thanks!